x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -421,48 +408,82 @@ PyTorchResultProcessor getResultProcessor() {
             return resultProcessor;
         }
 
-        synchronized void startProcess() {
-            process.set(pyTorchProcessFactory.createProcess(task, executorServiceForProcess, onProcessCrash()));
+        synchronized void startAndLoad(TrainedModelLocation modelLocation, ActionListener<Boolean> loadedListener) {
+            assert Thread.currentThread().getName().contains(UTILITY_THREAD_POOL_NAME)
+                : format("Must execute from [%s] but thread is [%s]", UTILITY_THREAD_POOL_NAME, Thread.currentThread().getName());
+
+            if (isStopped) {
+                logger.debug("[{}] start process is stopped", task.getModelId());
+                loadedListener.onFailure(new IllegalArgumentException("model loaded but process is stopped"));
```

Comment:
I think the message here is misleading as the process was stopped before we even got to try to load the model.